### PR TITLE
IALERT-3309 saml certs upload and validation

### PR DESCRIPTION
**Pull request by mau-synopsys**
- require_assertions_signed back to want_assertions_signed
- Remove enable check before validation of config model
- Endpoints for certs (encryption, signing, verification) file exists, upload
- Validation for cert files
- Code clean up to simplify some upload and verification calls
- Optional fields to nullable=true in liquibase files
- Added new fields to specify file path of cert files

### CODE REVIEW

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/validator/SAMLFileUploadValidator.java [hunk 1]
```diff
@@ -17,10 +18,9 @@
 import javax.xml.parsers.DocumentBuilder;
 import javax.xml.parsers.DocumentBuilderFactory;
 import javax.xml.parsers.ParserConfigurationException;
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.IOException;
-import java.io.InputStream;
+import java.io.*;
```
Comment (DanaMaxfield) [inline authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/validator/SAMLFileUploadValidator.java:21]: I believe we should be avoiding import * from a package.

**Review by DanaMaxfield — CHANGES_REQUESTED** [review]

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/validator/SAMLFileUploadValidator.java (hunk 1 above)
Comment (mau-synopsys) [reply authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/validator/SAMLFileUploadValidator.java]: Addressed!

**Review by DanaMaxfield — APPROVED** [review]

File: alert-database/src/main/resources/liquibase/6.13.0/concrete-saml-configuration-table.xml [hunk 2]
```diff
@@ -21,16 +21,16 @@
                 <constraints nullable="true"/>
             </column>
             <column name="enabled" type="BOOLEAN" defaultValueBoolean="false">
-                <constraints nullable="false"/>
```
Comment (psantos1113) [inline alert-database/src/main/resources/liquibase/6.13.0/concrete-saml-configuration-table.xml:24]: Why are all these fields now nullable?  If they are required fields for saml then the field shouldn't be nullable.

Comment (mau-synopsys) [reply alert-database/src/main/resources/liquibase/6.13.0/concrete-saml-configuration-table.xml:24]: These are the fields we discussed are required. 

Metadata url OR file path required
entityBaseUrl
entityid

I previously had everything set to false so corrected the non-required ones to false here

**Review by ChomickiM — APPROVED** [review]

File: alert-database/src/main/resources/liquibase/6.13.0/concrete-saml-configuration-table.xml (hunk 2 above)
Comment (psantos1113) [reply alert-database/src/main/resources/liquibase/6.13.0/concrete-saml-configuration-table.xml:24]: Enabled is required as well in the context of the database.  That cannot be null.

Comment (mau-synopsys) [reply alert-database/src/main/resources/liquibase/6.13.0/concrete-saml-configuration-table.xml]: Changed enabled, force_auth, and want_assertions_signed to nullable false

**Review by psantos1113 — APPROVED** [review]

**Pull request merged**